src/main/java/com/github/dariobalinzo/filter/WhitelistFilter.java

```diff
@@ -0,0 +1,30 @@
+package com.github.dariobalinzo.filter;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+public class WhitelistFilter implements Filter {
+
+    List<String> allowedValues;
```

Comment:
please make it `final private ` in order to make it immutable and avoid side-effects